static/js/components/inputs/DateField.js

```diff
@@ -95,11 +95,11 @@ export default class DateField extends React.Component {
       let validatedDay = Maybe.of(1);
       if (!omitDay) {
         validatedDay = validateDay(newEdit.day);
-        newEdit.day = mstr(validatedDay);
+        newEdit.day = dateForm(mstr(validatedDay), newEdit.day);
```

Comment:
I don't think you need this `dateForm` function here. 

When `newEdit.day` is a valid day number you'll get `Just(day)` back from `validateDay`, and otherwise you'll get `Nothing`. I'd suggest you use the `.map` method on `Maybe` to sort this out! Something like:

```js
newEdit.day = mstr(validatedDay.map(day => day === Number(newEdit.day) ? newEdit.day : day));
```

If `validatedDay` is `Nothing`, then calling `.map` on it will just return another `Nothing`, so we're safe to call `mstr` with it as an argument (`mstr(Nothing())` returns `''`).

If `validatedDay` is a day, but it's different than what the user typed in, then we want to only show them the validated value. If the user types in a bigger number than 31, for instance, we change it to be 31 (take a look at `lib/validation/date.js` for details).

If the value wrapped in `validatedDay` is equal to `Number(newEdit.day)` then we can safely let the user's input (with or without a leading zero) through.

does that make sense?